co_sim_io/impl/connection.hpp

```diff
@@ -55,14 +55,18 @@ class Connection
 
     ReturnInfo Connect()
     {
-        mpComm->Connect();
-        return ReturnInfo(); // TODO use this, should put a return code based on the bool that comes back from mpComm->Connect()
+        const bool is_connected = mpComm->Connect();
+        ReturnInfo ret_info;
+        ret_info.Set<bool>("connection_status", is_connected);
```

Comment:
I thought that you had an int with different options.

Reply:
sure, changed it :man_facepalming:

Reply:
I would also add some constexpr status values like:
- NotConnected: Connect is not called yet
- Connected: after successful call to Connect
- Disconnected: after successful call to Disconnect
- ConnectionError : Having a error message in return_info
- DisconnectionError: Having an error message in return_info

Reply:
How would this work for C?

Reply:
enums, or const int values

Reply:
Wouldn't it then make more sense to use enums also in C++?

Reply:
As you like, there is an implicit conversion from enum to int in both languages